### PR TITLE
initial setup and ci job for kubernetes-sigs/cluster-api-provider-openstack

### PR DESCRIPTION
**Pull request by dims**
We have a new repo : https://github.com/kubernetes-sigs/cluster-api-provider-openstack

This is for adding a new openstack external provider for [cluster-api](https://github.com/kubernetes-sigs/cluster-api)

Change-Id: I8d6637458cebe76e993590ab2767a350e1d53d9b
Signed-off-by: Davanum Srinivas <davanum@gmail.com>

### CODE REVIEW

**Comment by BenTheElder** [comment]
Sorry @dims, reviewed last night but missed submitting the comments, didn't realize I'd gotten into bulk review comment mode.

File: jobs/config.json [hunk 1]
```diff
@@ -11796,6 +11796,16 @@
       "sig-cluster-lifecycle"
     ]
   },
+  "pull-cluster-api-provider-openstack-make": {
+    "args": [
+      "make",
+      "images"
+    ],
+    "scenario": "execute",
```
[the remark this answers is not in the document]
Comment (dims) [reply jobs/config.json]: ack

File: prow/config.yaml [hunk 2]
```diff
@@ -3147,6 +3149,34 @@ presubmits:
         - "--repo=sigs.k8s.io/$(REPO_NAME)=$(PULL_REFS)"
         - "--root=/go/src"
         - "--upload=gs://kubernetes-jenkins/pr-logs"
+
+  kubernetes-sigs/cluster-api-provider-openstack:
```
[the remark this answers is not in the document]
Comment (dims) [reply prow/config.yaml]: ah even better. thanks

**Comment by BenTheElder** [comment]
`config/jobs/kubernetes-sigs/cluster-api-provider-openstack/config.yaml`, can't be config.yaml. the tooling is using the basname as a configmap key, it'll need a unique filename unfortunately.

**Pull request merged**